### PR TITLE
Indirect Reduction - qt Signal warnings

### LINKED ISSUE #24989: Indirect Data Reduction Signal warning
**Original reporter:** @gvardany 

### Description
The following warnings are seen in the terminal log when you load the Data Reduction interface:
image

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR removes unwanted run button signals coming from the IndirectDataReduction file. This file has no reason to be communicating with the run button.

**To test:**
1.`Interfaces`->`Indirect`->`Data Reduction`
2. Check the terminal log for any qt signal warnings

The Data Analysis and Tools interfaces should also be checked in the same way to make sure similar signal warnings are no longer present. Tools had the warning:

image

*No release notes required as these warnings were not in the previous release*

Fixes #24989

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/ConvFit.cpp [hunk 1]
```diff
@@ -161,6 +161,8 @@ std::string ConvFit::fitTypeString() const {
   return fitType;
 }
 
+void ConvFit::updateAvailableFitTypes() {}
```
Comment (gvardany) [inline qt/scientific_interfaces/Indirect/ConvFit.cpp]: I have hard time to understand why is this needed, can you explain these changes?

**Review by gvardany — COMMENTED** [review]
The warnings are gone, just a question about the other changes:

File: qt/scientific_interfaces/Indirect/ConvFit.cpp (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Indirect/ConvFit.cpp]: IndirectFitAnalysisTab was calling a slot called updateAvailableFitTypes, however ConvFit, MSDFit and IqtFit (which inherit from this) didn't have an updateAvailableFitTypes slot. This wasn't a good way of doing it so thank you for picking this up. Seeing as JumpFit is the only one that needs to update the available fit types, I have changed this so a signal is emitted instead by the IndirectFitAnalysisTab.

**Review by gvardany — APPROVED** [review]
Looks good
